### PR TITLE
Fix scrollwheel and lists

### PR DESCRIPTION
**Pull request by werfu**
Fix for issue #368 and fix a bug with ListBox crashing the app when having less than four items (out of bound seeking in a vector).

### CODE REVIEW

File: src/engine/localevent.cpp [hunk 1]
```diff
@@ -502,13 +508,12 @@ bool LocalEvent::HandleEvents( bool delay )
...
 
         // need for wheel up/down delay
-#if SDL_VERSION_ATLEAST( 2, 0, 0 )
-        if ( SDL_MOUSEWHEEL == event.type )
-            break;
-#else
-        if ( SDL_BUTTON_WHEELDOWN == event.button.button || SDL_BUTTON_WHEELUP == event.button.button )
-            break;
-#endif
+        #if SDL_VERSION_ATLEAST( 2, 0, 0 )
```
Comment (ihhub) [inline src/engine/localevent.cpp]: It's a good practice to use macro without indentation. I'll fix it :)

File: src/engine/localevent.cpp [hunk 2]
```diff
@@ -642,14 +645,12 @@ void LocalEvent::HandleMouseButtonEvent( const SDL_MouseButtonEvent & button )
...
-        case SDL_BUTTON_X2:
-#else
-        case SDL_BUTTON_WHEELDOWN:
-        case SDL_BUTTON_WHEELUP:
-#endif
-            mouse_rm = mouse_cu;
+#else 
+		case SDL_BUTTON_WHEELDOWN : 
+		case SDL_BUTTON_WHEELUP : 
+					mouse_rm = mouse_cu;
```
Comment (ihhub) [inline src/engine/localevent.cpp]: Please use 4 spaces instead of tabs :)

**Comment by werfu** [comment]
Sorry for the messy code, I installed VS C++ support and seems it doesn't come out of the box with code auto clean-up like I use in VSCode. What coding standard do you follow?

**Comment by ihhub** [comment]
Hi @werfu , it's okay. Such happens ;)
I've create a [wiki page](https://github.com/ihhub/fheroes2/wiki/Code-style) with a reference to clang-format. Surprisingly, it didn't detect styling issues :(

**Comment by ihhub** [comment]
Hi @werfu I tested your changes. Scroll speed is too high on SDL2. One turn on a mouse wheel moves the position of the list to the end or start.

**Comment by ihhub** [comment]
> Hi @werfu I tested your changes. Scroll speed is too high on SDL2. One turn on a mouse wheel moves the position of the list to the end or start.

I was writing about the list of maps.

**Comment by ihhub** [comment]
Hi @werfu , thank you very much for the contribution to the project!

**Pull request merged**